text/0730-semver-for-ts.md

```diff
@@ -0,0 +1,912 @@
+
+    This includes changing a previously type-and-value export such as `export class` to either—
+    
+    -   a type-only export, since the exported value symbol has been removed:
+
+        ```diff
+        -export class Foo {
+        +class Foo {
+          neato: string;
+        }
+        +
+        +export type { Foo };
+        ```
+    
+    -   a value-only export, since the exported type symbol has been removed:
+
+        ```diff
+        -export class Foo {
+        +class _Foo {
+          neato: string;
+        }
+        +
+        +export let Foo: typeof _Foo;
+        ```
+
+-   changing the kind (*value* vs. *type*) of an exported symbol in any way, since users' imports and own definitions may both be broken, since imports resolve all symbols imported together if they share a name:
+
+    -   Given a *value*-only exported symbol, including `namespace` declarations, adding a *type* export with the same name as the *value* may break users' code: they may have imported the value and safely created a type of the same name. Their existing import will now cause a re-declaration conflict. Note that this is distinct from adding an entirely new type export where there was no type or value export previously, since the user could never accidentally introduce the conflict, and could work around the conflict using the `as` import specifier when introducing the import.
+
+    -   Given a *type*-only exported symbol, including `type`, `interface`, or `export type` for a type or value, adding a *value* export with the same name may break users' code: they may have imported the type and safely created a value of the same name. Their existing import will now cause a re-declaration conflict. Note that this is distinct from adding an entirely new value export where there was no type or value export previously, since the user could never accidentally introduce the conflict, and could work around the conflict using the `as` import specifier when introducing the import.
+
+    -   Given a `namespace` export, changing it to a value-only export (that is, to an exported object) will break all nested type access, since types cannot be exported as nested members of a non-`namespace` values. (`namespace` exports cannot be directly converted to type-only exports.)
+
+-   changing an `interface` to a `type` alias will break any user code which used interface merging
+
+-   changing a `namespace` export to any other type will break any code which used namespace merging
+
+-   changing a `class` export to a type-only export will break any code which extended the class or constructed an instance of the class ([playground][class-to-type-only]), and changing a `class` export to a value-only export will break any code which referred to the class as a type ([playground][class-to-value-only])
```

Comment:
> These playground links suggest to me that this is observed rather than promised behavior. Is that the intended meaning? If so, should there be “as of TypeScript vX.Y” to anchor the observations?

I added a note in 68f2743 to clarify that it's a "sort of both" scenario here, since there is no spec for TS' type system—it's all implementation-defined, and I don't think that's likely ever to change (certainly not in the foreseeable future). I'll make a point to note the actual version for which these hold when the RFC is merged.